File: public/docs/_examples/architecture/ts/app/sales-tax.component.ts
// #docplaster
// #docregion
import {Component}       from 'angular2/core';
import {SalesTaxService} from './sales-tax.service';
import {TaxRateService}  from './tax-rate.service';

// #docregion metadata
// #docregion providers
@Component({
// #enddocregion providers
  selector:    'sales-tax',
  template: `
    <h2>Sales Tax Calculator</h2>
    Amount: <input #amountBox (change)="0">
    
    <div *ngIf="amountBox.value">
    The sales tax is
     {{ getTax(amountBox.value) | currency:'USD':true:'1.2-2' }}
    </div>
  `,
// #docregion providers
  providers:   [SalesTaxService, TaxRateService]
})
// #enddocregion providers
// #enddocregion metadata
/*
// #docregion metadata, providers
export class SalesTaxComponent { ... }
// #enddocregion metadata, providers
*/
// #docregion class
export class SalesTaxComponent {
// #docregion ctor
  constructor(private _salesTaxService: SalesTaxService) { }
// #enddocregion ctor

  getTax(value:string | number){
    return this._salesTaxService.getVAT(value);
  }
}
// #enddocregion class

File: public/docs/_examples/architecture/dart/lib/sales_tax_component.dart
import 'package:angular2/core.dart';

import 'sales_tax_service.dart';
import 'tax_rate_service.dart';

@Component(
    selector: 'sales-tax',
    template: '''
      <h2>Sales Tax Calculator</h2>
      Amount: <input #amountBox (change)="0">

      <div *ngIf="amountBox.value != ''">
      The sales tax is
       {{ getTax(amountBox.value) | currency:'USD':true:'1.2-2' }}
      </div>
    ''',
    providers: const [SalesTaxService, TaxRateService])
class SalesTaxComponent {
  SalesTaxService _salesTaxService;

  SalesTaxComponent(this._salesTaxService) {}

  num getTax(dynamic /* String | num */ value) =>
      this._salesTaxService.getVAT(value);
}
